codec/src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -0,0 +1,40 @@
+package io.netty.handler.codec.compression;
+
+import com.nixxcode.jvmbrotli.enc.BrotliOutputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+
+public class BrotliEncoder extends MessageToByteEncoder<ByteBuf> {
+
+    private final int totalLength;
+    private int writtenLength;
+
+    public BrotliEncoder(int totalLength) {
```

Comment:
How do you see totalLength working if there is a streaming response?

Reply:
`Content-Length` won't work?